In [2]:
#Import libraries and necesary metrics
import sklearn
from sklearn import metrics
from sklearn.metrics import roc_auc_score
from sklearn.metrics import accuracy_score
import pandas as pd
import numpy
import numpy as np

In [52]:
#FPR and TPR
cols_list = ["lightleak", "pen", "ruler", "border", "corner", "blur", "hair_some", "hair_obscure"]
#NOTE: DO NOT FORGET TO MANUALLY ADD THE FIRST COLUMN (IMAGEIDS) AND FIRST ROW (LABELS) TO THE FINAL CSV FILE BEFORE RUNNING THIS
dfpred = pd.read_csv("/Volumes/J_Bac/2021/Results/oofv2_2021_08-15(2)_Final.csv", usecols=cols_list)
dftrue = pd.read_csv("./QA_Multi_Temp_Final.csv", usecols=cols_list)
for feature in cols_list:
    y_pred = list(dfpred[feature])
    y_true = list(dftrue[feature])
    #auc = metrics.roc_auc_score(y_true, y_pred)
    rocauc = roc_auc_score(y_true, y_pred)
    print('ROC for %s = %f'%(feature,rocauc))

ROC for lightleak = 0.971448
ROC for pen = 0.994216
ROC for ruler = 0.993112
ROC for border = 0.994204
ROC for corner = 0.993128
ROC for blur = 0.817548
ROC for hair_some = 0.984872
ROC for hair_obscure = 0.995914


# 15Aug-Run2
ROC for lightleak = 0.971448
ROC for pen = 0.994216
ROC for ruler = 0.993112
ROC for border = 0.994204
ROC for corner = 0.993128
ROC for blur = 0.817548
ROC for hair_some = 0.984872
ROC for hair_obscure = 0.995914

# 15Aug-Run1  -> Increased persistance, corrected cropping and set the minimum blur to 3 ([3,30] range)
ROC for lightleak = 0.977422
ROC for pen = 0.992386
ROC for ruler = 0.993281
ROC for border = 0.994392
ROC for corner = 0.992394
ROC for blur = 0.819753
ROC for hair_some = 0.985599
ROC for hair_obscure = 0.996411

# 14Aug-Run2
ROC for lightleak = 0.979799
ROC for pen = 0.994471
ROC for ruler = 0.992574
ROC for border = 0.995200
ROC for corner = 0.992420
ROC for blur = 0.804636
ROC for hair_some = 0.984839
ROC for hair_obscure = 0.995204

# 14Aug-Run1 
ROC for lightleak = 0.918490
ROC for pen = 0.982030
ROC for ruler = 0.939359
ROC for border = 0.991419
ROC for corner = 0.891795
ROC for blur = 0.797552
ROC for hair_some = 0.980713
ROC for hair_obscure = 0.994218

In [4]:
#FPR and TPR (per feature)
cols_list = ["lightleak", "pen", "ruler", "border", "corner", "blur", "hair_some", "hair_obscure"]
dfpred = pd.read_csv("/Volumes/J_Bac/2021/Results/oofv2_2021_08-14(2)_Final.csv", usecols=cols_list)
dftrue = pd.read_csv("/Volumes/J_Bac/2021/QA_Multi_Temp_Final.csv", usecols=cols_list)
df3 = pd.DataFrame(columns=cols_list, index = range(100))
x = 'blur'
y_pred = list(dfpred[x])
y_true = list(dftrue[x])
fpr, tpr, thresholds = sklearn.metrics.roc_curve(y_true, y_pred)
for feature in cols_list:
    for i in range(1641): #Change for each feature
        df3.loc[i, x]=tpr[i]
print(df3)
#df3.to_csv('/Volumes/J_Bac/2021/Results/tpr_8-14(2).csv')

     lightleak  pen ruler border corner         blur hair_some hair_obscure
0          NaN  NaN   NaN    NaN    NaN            0       NaN          NaN
1          NaN  NaN   NaN    NaN    NaN  0.000469925       NaN          NaN
2          NaN  NaN   NaN    NaN    NaN  0.000469925       NaN          NaN
3          NaN  NaN   NaN    NaN    NaN   0.00234962       NaN          NaN
4          NaN  NaN   NaN    NaN    NaN   0.00234962       NaN          NaN
...        ...  ...   ...    ...    ...          ...       ...          ...
1636       NaN  NaN   NaN    NaN    NaN      0.99953       NaN          NaN
1637       NaN  NaN   NaN    NaN    NaN      0.99953       NaN          NaN
1638       NaN  NaN   NaN    NaN    NaN      0.99953       NaN          NaN
1639       NaN  NaN   NaN    NaN    NaN            1       NaN          NaN
1640       NaN  NaN   NaN    NaN    NaN            1       NaN          NaN

[1641 rows x 8 columns]


In [4]:
#Roc-Auc New
cols_list = ["lightleak", "pen", "ruler", "border", "corner", "blur", "hair_some", "hair_obscure"]
for feature in cols_list:
    y_pred = pd.read_csv("/Volumes/J_Bac/Falses/output.csv", usecols=[feature])
    y_true = pd.read_csv("/Volumes/J_Bac/Falses/QA_Multi_Temp_Final.csv", usecols = [feature])
    rocauc = roc_auc_score(y_true, y_pred)
    print(feature, rocauc)

lightleak 0.9569533972125436
pen 0.9799412689965676
ruler 0.9785060345404499
border 0.9961713827519578
corner 0.985083836630493
blur 0.8221944758216879
hair_some 0.9676680351751759
hair_obscure 0.9632538473647172


In [7]:
#Roc-Auc Original
cols_list = ["lightleak", "pen", "ruler", "border", "corner", "blur", "hair_some", "hair_obscure"]
for feature in cols_list:
    y_pred = pd.read_csv("/Volumes/J_Bac/Falses/output.csv", usecols=[feature])
    y_true = pd.read_csv("/Volumes/J_Bac/Falses/QA_Multi_Temp.csv", usecols = [feature])
    rocauc = roc_auc_score(y_true, y_pred)
    print(feature, rocauc)

lightleak 0.9564135115624923
pen 0.9597155593772628
ruler 0.9526238327745653
border 0.9774620962803814
corner 0.9697537940935193
blur 0.7866478543673554
hair_some 0.9439447209213527
hair_obscure 0.9420357693132837


In [6]:
#Accuracy (Currently not working)
cols_list = ["lightleak", "pen", "ruler", "border", "corner", "blur", "hair_some", "hair_obscure"]
for feature in cols_list:
    y_pred = pd.read_csv("/Volumes/J_Bac/Falses/output.csv", usecols=[feature])
    y_true = pd.read_csv("/Volumes/J_Bac/Falses/QA_Multi_Temp_Final.csv", usecols=[feature])
    #accuracy = accuracy_score(y_true, y_pred)
    #print(feature, accuracy)